dbschema/src/main/java/org/folio/rest/persist/ddlgen/Index.java

```diff
@@ -83,24 +84,45 @@ public String getFinalSqlExpression(String tableLoc) {
           result .append(" , ");
         }
         String [] rawExpandedTerm = splitIndex[i].split("\\.");
-        StringBuilder expandedTerm = formatExpandedTerm(rawExpandedTerm);
-        result.append(tableLoc).append(".jsonb").append(expandedTerm);
+        StringBuilder expandedTerm = formatExpandedTerm(tableLoc + ".jsonb",rawExpandedTerm);
+        result.append(expandedTerm);
       }
       result.append(")");
       return result.toString();
     }
 
   }
-  private StringBuilder formatExpandedTerm(String[] rawExpandedTerm) {
-    StringBuilder expandedTerm = new StringBuilder();
 
+  private StringBuilder formatExpandedTerm(String table, String[] rawExpandedTerm) {
+    StringBuilder expandedTerm = new StringBuilder();
+    StringBuilder result = new StringBuilder();
```

Comment:
Creating additional StringBuilders for substrings should be avoided. The idea of a StringBuilder is to try to append to a single result StringBuilder.

Reply:
I understand what you are saying regarding this stringbuilder, however the information needed to do this wrapping comes naturally while iterating through the list of terms.  But the actual function call itself needs to wrap the term.  So I feel like my options are to use a sub builder like I did, or add a bunch of logic to detect the start and stop of a term with the brackets in it.